services/getstream/followTopic.js

```diff
@@ -96,28 +97,14 @@ const addTopicToChatTab = async (
     });
 
     console.log('prepare follow channel', topic);
-    let textOwnUser = 'You joined this community';
-    if (isAnonymous) textOwnUser += ' incognito';
 
     await channel.create();
     await channel.addMembers([userId]);
     if (withSystemMessage) {
       console.log('system message sent');
-      await channel.sendMessage(
-        {
-          user_id: userId,
-          text: textOwnUser,
-          other_text: '',
-          own_text: textOwnUser,
-          system_user: userId,
-          isSystem: true,
-          type: 'system',
-          only_show_to_system_user: true
-        },
-        {
-          skip_push: true
-        }
-      );
+      await Getstream.chat.sendFollowTopicSystemMessage(channel, userId, {
+        isJoinedAnonymous: isAnonymous
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `console.log` for debugging purposes should be replaced with a proper logging library that can provide different levels of logging (info, debug, error, etc.) and can be configured to output logs based on the environment (development, production, etc.). This will help in better log management and avoid exposing sensitive information in production.

Also, there's no error handling for the async operations `channel.create()`, `channel.addMembers([userId])`, and `Getstream.chat.sendFollowTopicSystemMessage(channel, userId, {isJoinedAnonymous: isAnonymous})`. If any of these operations fail, the error will be unhandled, which could lead to unexpected behavior. Consider wrapping each operation in a try-catch block or using Promise.allSettled to handle potential rejections.

Lastly, the `withSystemMessage` condition is checked after the channel creation and adding members. If `withSystemMessage` is false, these operations are unnecessary. Consider checking this condition earlier to avoid unnecessary operations.
```diff
-    console.log('prepare follow channel', topic);
-    await channel.create();
-    await channel.addMembers([userId]);
-    if (withSystemMessage) {
-      console.log('system message sent');
-      await Getstream.chat.sendFollowTopicSystemMessage(channel, userId, {
-        isJoinedAnonymous: isAnonymous
-      });
+    if (withSystemMessage) {
+      console.log('prepare follow channel', topic);
+      try {
+        await channel.create();
+        await channel.addMembers([userId]);
+        console.log('system message sent');
+        await Getstream.chat.sendFollowTopicSystemMessage(channel, userId, {
+          isJoinedAnonymous: isAnonymous
+        });
+      } catch (error) {
+        console.error('Error while following topic:', error);
+      }
     }
     console.log('channel followed');
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->